romdisk/uart.asm

```diff
@@ -122,23 +124,47 @@ uartsnd_main:
         ld l, a
         ; HL contains the string to output on the UART
         ; Get its length and save it in BC
-        push hl
-        call strlen
+        
+        ; Filename in HL
+        ld b, h
+        ld c, l
+        ld h, O_RDONLY
+        OPEN()
+        or a
+        jp m, uart_file_error
+        
+        ld h, a
+        ld de, 0x8000
+        ld bc, 16384
+        READ()
+        ; Will return the size of the file buffer, saved in BC
         push bc
+        CLOSE()
```

Comment:
If the file is bigger than `16384` bytes, the rest of the bytes will be ignored and not send.

Reply:
Fixed.